### ISSUE TITLE
F5 test failing in base_ami job

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#8908]
F5 test is failing in base_ami: https://ci.openshift.redhat.com/jenkins/job/base_ami/1912

```
W0517 03:20:22.687327   23278 f5.go:255] Strict certificate verification is *DISABLED*
E0517 03:20:22.688320   23278 f5.go:758] partition path "/Common" error: Encountered an error on GET request to URL https://127.0.0.1:51572/mgmt/tm/sys/folder/~Common: error: client.Do failed: Get https://127.0.0.1:51572/mgmt/tm/sys/folder/~Common: dial tcp 127.0.0.1:51572: getsockopt: connection refused
--- FAIL: TestInitializeF5Plugin (0.01s)
	plugin_test.go:1265: Failed to initialize test router: Failed to create new F5 router: Encountered an error on GET request to URL https://127.0.0.1:51572/mgmt/tm/sys/folder/~Common: error: client.Do failed: Get https://127.0.0.1:51572/mgmt/tm/sys/folder/~Common: dial tcp 127.0.0.1:51572: getsockopt: connection refused
W0517 03:20:22.704080   23278 f5.go:255] Strict certificate verification is *DISABLED*
E0517 03:20:22.706095   23278 f5.go:758] partition path "/Common" error: Encountered an error on GET request to URL https://127.0.0.1:60265/mgmt/tm/sys/folder/~Common: error: client.Do failed: Get https://127.0.0.1:60265/mgmt/tm/sys/folder/~Common: dial tcp 127.0.0.1:60265: getsockopt: connection reset by peer
--- FAIL: TestF5RouterPartition (0.02s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x60 pc=0x4d55e9]

goroutine 14 [running]:
panic(0x1375500, 0xc82000e130)
	/usr/lib/golang/src/runtime/panic.go:481 +0x3ff
testing.tRunner.func1(0xc8203333b0)
	/usr/lib/golang/src/testing/testing.go:467 +0x22a
panic(0x1375500, 0xc82000e130)
	/usr/lib/golang/src/runtime/panic.go:443 +0x521
github.com/openshift/origin/pkg/router/f5.(*mockF5).close(0x0)
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/router/f5/plugin_test.go:261 +0x39
github.com/openshift/origin/pkg/router/f5.TestF5RouterPartition(0xc8203333b0)
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/router/f5/plugin_test.go:1408 +0x1a1
testing.tRunner(0xc8203333b0, 0x1e29058)
	/usr/lib/golang/src/testing/testing.go:473 +0xdd
created by testing.RunTests
	/usr/lib/golang/src/testing/testing.go:582 +0xae3
FAIL	github.com/openshift/origin/pkg/router/f5	0.078s
```

**Comment by smarterclayton** [comment]
@Miciah we just switched to Go 1.6 to run unit tests - not sure whether that's related or not.

**Comment by Miciah** [comment]
I'm eyeing lines 1408 and 1418 at https://github.com/openshift/origin/commit/37e1e96d36670f4134c07cfc04b8a774bc718c7d#diff-e6c049c643b77b500bba15d06c8f75e3R1408.  It does look like a double-close, but I'm confused why we see a panic on the *first* close.  @ramr, thoughts?

**Comment by smarterclayton** [comment]
Is mockF5 returning an error?  In which case the thing you're invoking
close() on is nil?

**Comment by Miciah** [comment]
Yeah.  There are two problems in the code I pointed to: (1) it does a `close` even if `newTestRouter` returns an error (in which case `mockF5` will be `nil`) and (2) it has a double-close.  As to why `newTestRouter` is failing, my best guess is that, problem (3), we create new test routers in `TestF5RouterPartition` without closing them, and maybe that causes us to run up against some limit that causes new `httptest` servers to fail to create.  I'll make a PR to fix these 3 problems.

**Comment by Miciah** [comment]
I meant `TestInitializeF5Plugin`, not `TestF5RouterPartition`.

**Comment by Miciah** [comment]
Per https://github.com/openshift/origin/issues/4417, we can probably drop the funky stuff we do to avoid ECDHE ciphers.  Funky stuff is always suspect.

**Comment by smarterclayton** [comment]
We in origin only now use Go 1.6.

**Comment by Miciah** [comment]
To be clear, the funky stuff is only in the test code.  It is a workaround to configure the test server not to use ECDHE ciphers.  We added it because the tests were slow otherwise (they would eventually succeed, but they took long enough to cause timeouts in Travis, IIRC).  Is there a problem with deleting the workaround in the test code, or do we need tests to run fast on older Go?

**Comment by smarterclayton** [comment]
We can delete the code - we are dropping the older Go.

**Comment by liggitt** [comment]
+1 for dropping, I verified performance wasn't an issue any more

**Comment by miminar** [comment]
Resolved by #8925
